stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocket.kt

```diff
@@ -297,8 +302,8 @@ internal open class ChatSocket constructor(
 
     private fun callListeners(call: (SocketListener) -> Unit) {
         synchronized(listeners) {
-            listeners.forEach { listener ->
-                eventUiHandler.post { call(listener) }
+            eventUiHandler.post {
+                listeners.forEach(call)
             }
         }
```

Comment:
`synchronized` should be moved into the thread where iteration runs, otherwise, it won't work:
```kotlin
        eventUiHandler.post {
            synchronized(listeners) {
                listeners.forEach(call)
            }
        }
```

Reply:
Fixed